libvast/test/command.cpp

```diff
@@ -111,41 +108,20 @@ TEST(full name) {
   CHECK_EQUAL(cmd2->full_name(), "foo bar");
 }
 
-TEST(parsing value) {
-  auto cmd = root.add<foo>("foo");
-  exec("foo -v 42");
-  CHECK_EQUAL(cmd->flag, false);
-  CHECK_EQUAL(cmd->value, 42);
-  CHECK_EQUAL(caf::deep_to_string(options),
-              R"([("flag", false), ("value", 42)])");
-}
-
-TEST(parsing flag) {
-  auto cmd = root.add<foo>("foo");
-  exec("foo --flag");
-  CHECK_EQUAL(cmd->flag, true);
-  CHECK_EQUAL(cmd->value, 0);
-  CHECK_EQUAL(caf::deep_to_string(options),
-              R"([("flag", true), ("value", 0)])");
-}
-
-TEST(parsing both) {
-  auto cmd = root.add<foo>("foo");
+TEST(parsing args) {
+  root.add<foo>("foo");
   exec("foo --flag -v 42");
-  CHECK_EQUAL(cmd->flag, true);
-  CHECK_EQUAL(cmd->value, 42);
-  CHECK_EQUAL(caf::deep_to_string(options),
-              R"([("flag", true), ("value", 42)])");
+  CHECK_EQUAL(get_or(options, "flag", false), true);
+  CHECK_EQUAL(get_or(options, "value", 0), 42);
 }
 
 TEST(nested arg parsing) {
   auto cmd1 = root.add<foo>("foo");
-  auto cmd2 = cmd1->add<bar>("bar");
+  cmd1->add<bar>("bar");
   exec("foo -v 42 bar -o 123");
-  CHECK_EQUAL(cmd1->value, 42);
-  CHECK_EQUAL(cmd2->other_value, 123);
-  CHECK_EQUAL(caf::deep_to_string(options),
-              R"([("flag", false), ("other-value", 123), ("value", 42)])");
+  CHECK_EQUAL(get_or(options, "flag", false), false);
+  CHECK_EQUAL(get_or<integer>(options, "value", 0), 42);
```

Comment:
Shouldn't that work without the explicit `<integer>`?

Reply:
You are right, `<integer>` is redundant.